clients/src/main/java/org/apache/kafka/common/requests/ProduceResponse.java

```diff
@@ -223,6 +269,18 @@ protected Struct toStruct(short version) {
                 if (partStruct.hasField(LOG_APPEND_TIME_KEY_NAME))
                     partStruct.set(LOG_APPEND_TIME_KEY_NAME, part.logAppendTime);
                 partStruct.setIfExists(LOG_START_OFFSET_FIELD, part.logStartOffset);
+
```

Comment:
nit: we can probably do some cleanup in pr.3, e.g. line 269 above can be replaced with `setIfExist` as well.

Reply:
cc @tuvtran

Reply:
ack :)